### PR TITLE
Clean up go module after generating codes

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
The code-generator adds temporary module to go.mod when generating codes. So we should clean up go.mod.

```shell
diff --git a/go.mod b/go.mod
index 806cc12..4e11427 100644
--- a/go.mod
+++ b/go.mod
@@ -64,6 +64,7 @@ require (
        github.com/stretchr/testify v1.8.2 // indirect
        go.uber.org/atomic v1.11.0 // indirect
        go.uber.org/multierr v1.11.0 // indirect
+       golang.org/x/crypto v0.1.0 // indirect
        golang.org/x/mod v0.10.0 // indirect
        golang.org/x/net v0.10.0 // indirect
        golang.org/x/oauth2 v0.6.0 // indirect
diff --git a/go.sum b/go.sum
index ead83fc..d79d991 100644
--- a/go.sum
+++ b/go.sum
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by tenzen-y** [comment]
@alculquicondor ping

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -43,3 +43,6 @@ then
   fi
   mv sigs.k8s.io/kueue/client-go ${KUEUE_ROOT}/client-go
 fi
+
+# We need to clean up the go.mod file since code-generator adds temporary library to the go.mod file.
+"${GO_CMD}" mod tidy
```
Comment (alculquicondor) [inline hack/update-codegen.sh:48]: should we add it to the tools package instead?

Comment (tenzen-y) [reply hack/update-codegen.sh:48]: Ah, right.

Comment (tenzen-y) [reply hack/update-codegen.sh:48]: Uhm, tools don't seem work well.

```shell
go: finding module for package golang.org/x/crypto
sigs.k8s.io/kueue/hack/internal/tools imports
        golang.org/x/crypto: module golang.org/x/crypto@latest found (v0.10.0), but does not contain package golang.org/x/crypto
```

Comment (alculquicondor) [reply hack/update-codegen.sh:48]: :thinking: 

no idea

/lgtm
/approve

**Pull request merged**